### PR TITLE
Print a more user-friendly error message for leftover diff marks

### PR DESCRIPTION
**Pull request by ISSOtm**

### CODE REVIEW

File: src/asm/parser.y [hunk 1]
```diff
@@ -690,17 +690,19 @@ asmfile		: lines
...
 ;
 
-plain_directive	: label
-		| label cpu_command
-		| label macro
-		| label directive
-		| assignment_directive
+opt_diff_mark	: %empty // OK
+		| T_OP_ADD {
+			error("syntax error: '+' at the beginning of the line (this looks like a leftover diff mark)");
```
Comment (Rangi42) [inline src/asm/parser.y:701]: To match bison's own syntax error messages:
```suggestion
			error("syntax error, unexpected '+' (this looks like a leftover diff mark)");
```

File: src/asm/parser.y [hunk 2]
```diff
@@ -690,17 +690,19 @@ asmfile		: lines
...
+		}
+		| T_OP_SUB {
+			error("syntax error: '-' at the beginning of the line (this looks like a leftover diff mark)");
```
Comment (Rangi42) [inline src/asm/parser.y:704]:
```suggestion
			error("syntax error, unexpected '-' (this looks like a leftover diff mark)");
```

File: src/asm/parser.y (hunk 1 above)
Comment (Rangi42) [reply src/asm/parser.y:701]: For in case they write `+    ld a, 2 + 2`? Sure.

**Comment by ISSOtm** [comment]
Tried a new one. What do you think?

**Review by Rangi42 — APPROVED** [review]

**Pull request merged**